### PR TITLE
Feature/string catalogs

### PR DESCRIPTION
**Pull request by andrija-ostojic**
## Summary
This PR adds support for String Catalogs as a localization source

**Related issue**: 

[Productive task with a longer description of the feature](https://app.productive.io/1-infinum/tasks/17778177)
### Type

- [x] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [ ] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

#### Additional information

- [ ] This pull request introduces a **breaking change**.

### Description
- Example project is updated to use String catalogs. Localized strings are divided into 2 catalogs in the project, to showcase that multiple catalogs are supported.

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have tested my changes, including edge cases.
- [ ] I have added necessary tests for the changes introduced (if applicable).
- [ ] I have updated the documentation to reflect my changes (if applicable).

### CODE REVIEW

File: SwiftI18n/Classes/Main/I18nManager.swift [hunk 1]
```diff
@@ -45,7 +51,26 @@ public class I18nManager {
...
+        if let localizationPerformingCustomBlock {
+            return localizationPerformingCustomBlock(key, language, bundle)
+        }
+
+        switch localizationSource {
+        case .stringsFiles:
+            return NSLocalizedString(key, tableName: language, bundle: bundle, comment: "")
+        case .stringCatalogs(let catalogs):
+            guard let path = bundle.path(forResource: language, ofType: Constants.lprojExtension), let languageBundle = Bundle(path: path) else {
+                return NSLocalizedString(key, tableName: language, bundle: bundle, comment: "")
```
Comment (nikolamajcen) [inline SwiftI18n/Classes/Main/I18nManager.swift]: When the `.lproj` directory can't be resolved, this falls back to `NSLocalizedString(... tableName: language ...)`. In a String-Catalogs-only project that table doesn't exist, so this always returns `key` (effectively a miss). Either drop the fallback (return `key` directly) or add a comment explaining why it's preserved. Right now it reads like dead code.

File: SwiftI18n/Classes/Main/I18nManager.swift [hunk 2]
```diff
@@ -45,7 +51,26 @@ public class I18nManager {
...
+            }
+            for catalog in catalogs {
+                let localizedString = languageBundle.localizedString(forKey: key, value: Constants.keyNotFoundFallbackValue, table: catalog)
+                // If the specified key is not found, `value` parameter will be returned instead so we compare to it to check if the key was found
+                if localizedString != Constants.keyNotFoundFallbackValue {
+                    return localizedString
+                }
+            }
+            return key
```
Comment (nikolamajcen) [inline SwiftI18n/Classes/Main/I18nManager.swift:89]: Two points here:

**a) Document that catalog order is semantically meaningful.** If the same key exists in two catalogs, the array order in `.stringCatalogs(catalogs: [...])` decides which wins. This belongs on the enum case's `///` doc.

**b) O(catalogs) per lookup.** `Bundle.localizedString` caches the parsed table internally, so each probe is one dictionary read, but it still scales linearly with `catalogs.count` on every `loc[key]` call (table cells, list rows).

Cheap optimization: cache the last winning catalog per key, then probe it first.

```swift
private var catalogHitCache: [String: String] = [:]
private let catalogCacheLock = NSLock()

// in the .stringCatalogs branch:
let probeOrder: [String] = {
    catalogCacheLock.lock(); defer { catalogCacheLock.unlock() }
    if let cached = catalogHitCache[key] {
        return [cached] + catalogs.filter { $0 != cached }
    }
    return catalogs
}()
for catalog in probeOrder {
    let result = languageBundle.localizedString(forKey: key, value: Constants.keyNotFoundFallbackValue, table: catalog)
    if result != Constants.keyNotFoundFallbackValue {
        catalogCacheLock.lock(); catalogHitCache[key] = catalog; catalogCacheLock.unlock()
        return result
    }
}
return key
```

Invalidate the cache in the `language` setter and on `localizationSource` change (`didSet`). If you'd rather not add the cache, document the O(catalogs) cost on `.stringCatalogs` so adopters can decide.

File: SwiftI18n/Classes/Main/I18nManager.swift [hunk 3]
```diff
@@ -17,6 +22,7 @@ public class I18nManager {
     public var defaultLanguage: String?
     public var availableLanguages: [String]?
     public var bundle: Bundle = .main
+    public var localizationSource: LocalizationSource = .stringsFiles
```
Comment (nikolamajcen) [inline SwiftI18n/Classes/Main/I18nManager.swift:34]: Defaulting to `.stringsFiles` preserves backwards compatibility, which is good. Please add a `///` noting this and that it should be set before the first localization lookup (the example app's AppDelegate reorder is a hint that this matters).

File: SwiftI18n/Classes/Main/I18nManager.swift [hunk 4]
```diff
@@ -45,7 +51,26 @@ public class I18nManager {
     /// Returns a localized string for the given key and language.
     /// If ``localizationPerformingCustomBlock`` is set, it delegates to that closure, otherwise falls back to `NSLocalizedString`
     public func localizationPerformingBlock(_ key: String, _ language: String, _ bundle: Bundle = .main) -> (String) {
```
Comment (nikolamajcen) [inline SwiftI18n/Classes/Main/I18nManager.swift:68]: Re: `localizationPerformingCustomBlock` precedence. The custom block now takes precedence over `localizationSource` (matches its old precedence over `NSLocalizedString`). The existing doc on `localizationPerformingCustomBlock` says "overrides the default localization lookup," which is now ambiguous with two defaults. Please update that doc comment to state the precedence explicitly: custom block > `localizationSource`.

File: SwiftI18n/Classes/Main/LocalizationSource.swift [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+//  LocalizationSource.swift
+//  Pods
+//
+//  Created by Andrija Ostojic on 12. 5. 2026..
+//
+
+public enum LocalizationSource {
+    case stringsFiles
+    case stringCatalogs(catalogs: [String])
+}
```
Comment (nikolamajcen) [inline SwiftI18n/Classes/Main/LocalizationSource.swift:28]: The rest of the manager is well-documented, so this new public API needs the same treatment. Please add `///` docs to:

- The enum itself: what it controls, default value, where to set it.
- `.stringsFiles`: preserves the legacy `NSLocalizedString` table-per-language behavior.
- `.stringCatalogs(catalogs:)`: catalog names are `.xcstrings` filenames without extension; array order matters when keys overlap; catalogs must be present in the main bundle.

File: Example/SwiftI18n/AppDelegate.swift [hunk 6]
```diff
@@ -20,8 +20,8 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil
     ) -> Bool {
 
-        setupViewHierarchy()
         setupDefaultLanguage()
+        setupViewHierarchy()
```
Comment (nikolamajcen) [inline Example/SwiftI18n/AppDelegate.swift:24]: Swapping `setupViewHierarchy()` and `setupDefaultLanguage()` implies views now read localized strings during construction and require `localizationSource` to be configured first. Two suggestions:

1. Call this ordering requirement out in the README. The failure mode (untranslated keys appearing in UI) is silent.
2. Consider a debug-only `assertionFailure` or `os_log` inside `I18nManager` when `.stringCatalogs` is set but the catalog can't be found, to surface misconfiguration loudly during development.

**Review by nikolamajcen — COMMENTED** [review]
Initial pass on the String Catalogs change. Inline comments below cover the library API and the example wiring. A few higher-level notes follow as separate comments (tests, deleted .strings files, podspec/README/release).

**Comment by nikolamajcen** [comment]
**Tests**: the checklist item is unchecked and `Example/Tests` already exists. The new branching (lproj resolution, catalog iteration, sentinel comparison, fallback path) is worth covering with:

- key resolved from first listed catalog
- fall-through to second catalog when key isn't in the first
- unknown key returns `key`
- `localizationPerformingCustomBlock` still takes precedence when `.stringCatalogs` is set
- fallback-language path through `.stringCatalogs` (second call inside `localizedString(forKey:)`)

**Comment by nikolamajcen** [comment]
**Deleted `.strings` files in `Example/SwiftI18n/LocalizedStrings/`**: pre-existing nit, `hr.strings` was incorrectly placed under `en.lproj/`. Since you're cleaning this up, also consider keeping a minimal `.strings`-based example (or a README note in `Example/`) so adopters can see both setup styles. `.stringsFiles` is still a supported code path and the example app no longer demonstrates it.

**Comment by nikolamajcen** [comment]
**Toolchain requirement, podspec, README, CHANGELOG**

**Xcode 15+ requirement.** `SwiftI18n.podspec` keeps `deployment_target = '13.0'`, which is correct. `.xcstrings` compile down to per-language `.strings`/`.stringsdict` at build time, so the runtime API used here works on every supported iOS version. What's new is that authoring `.xcstrings` requires **Xcode 15+**. Adopters on older Xcode picking up this version and trying to use `.stringCatalogs` will find their catalog files don't compile. Mention this in the README section that documents `.stringCatalogs`.

**Release hygiene.** This PR adds new public API (`LocalizationSource`, `I18nManager.localizationSource`) but doesn't touch the podspec, README, or release notes:

- **Podspec**: bump `2.0.1` to `2.1.0` (non-breaking feature, defaults preserved, minor bump per SemVer).
- **README**: add a "Using String Catalogs" subsection covering the AppDelegate snippet, the filename-without-extension convention, the ordering rule when keys overlap, and the Xcode 15+ requirement.
- **CHANGELOG**: repo has no `CHANGELOG.md`; release notes appear to live on GitHub Releases. Either add an entry there with the release, or start a `CHANGELOG.md` with this PR (maintainer's call).

README + podspec bump should land before this leaves draft.

File: SwiftI18n/Classes/Main/I18nManager.swift (hunk 2 above)
Comment (andrija-ostojic) [reply SwiftI18n/Classes/Main/I18nManager.swift:89]: I decided to document everything, but skipped catalog lookup optimization because it looks to be a premature optimization. The number of catalogs is not likely to be big, if ever there is more than one in the bundle, so I think we are pretty safe with this approach for now

**Comment by andrija-ostojic** [comment]
Just to address a couple of the comments :
- Regarding tests, the library has to tests at all at the moment so I would tackle that in a separate PR and ignore tests for this one.
- I don't think we need both examples in the Example project, having String Catalogs example is enough
- I bumped the version to 2.1.0
